### ISSUE TITLE
Build Fails Without WITH_DISNEY_MATERIAL

### ISSUE THREAD

**Issue by danieldresser-ie** [appleseedhq/appleseed#1495]
In tag for 1.7.0-beta, I get an undefined reference error during build if I build without WITH_DISNEY_MATERIAL:
```
[ 93%] Linking CXX executable appleseed.cli
../appleseed/libappleseed.so: undefined reference to `renderer::DisneyMaterialFactory::get_model() const'
../appleseed/libappleseed.so: undefined reference to `vtable for renderer::DisneyMaterialFactory'
collect2: error: ld returned 1 exit status
```

From a quick investigation, the likely culprit appears to be this:

https://github.com/appleseedhq/appleseed/blob/a9c373b49117a4e61bec1c1d7499560b59dc0d2a/src/appleseed/renderer/modeling/project/projectfileupdater.cpp#L1348

The project file updater appears to need to be able to create a disney material for comparison, and I don't see any ifdef guards on it, which seems like it could cause a reference to an undefined symbol when we're not compiling DisneyMaterial.

Not sure whether this is important to us at ImageEngine - we should maybe just get the seexpr dependency set up right - but I guess if the compile option is there, and off by default, it should probably at least compile with it off.

**Comment by dictoon** [comment]
Hey Daniel, thanks for the report, we will fix it in master and will report the fix in the 1.7.0 branch.

Thanks again!

**Comment by dictoon** [comment]
Fixed in master and in 1.7.0-beta branch: https://github.com/appleseedhq/appleseed/tree/1.7.0-beta
